plugins/kubernetes-backend/src/service/getKubernetesObjectsForServiceHandler.ts

```diff
@@ -45,8 +46,28 @@ const DEFAULT_OBJECTS = new Set<KubernetesObjectTypes>([
   'ingresses',
 ]);
 
+function parseLabelSelector(entity: ComponentEntityV1alpha1): string {
+  if (
```

Comment:
you can also do

```ts
const matchLabels = entity?.spec?.kubernetes?.selector?.matchLabels;
if (matchLabels) {...
```

(but it's pretty defensive to assume that the caller might pass in anything except a `ComponentEntityV1alpha1` :))